Changing Debug contexts breaks Disassembly view

**Describe the bug**

The disassembly view stops working if the debug context changes. This can be exceptions (see the end of report for examples) or simply corrupt displaying in the view

**To Reproduce**
Steps to reproduce the behavior:

1. Start two debug sessions
2. Open disassembly view
3. Switch between frames in the two sessions
4. Observe exceptions in the error log

**Expected behavior**

Disassembly works with multiple sessions.

**Screenshots**

Example of the view broken:

![image](https://github.com/eclipse-cdt/cdt/assets/679236/42789763-0933-4a0b-8d49-8eb86c0b4c1d)

**Version Information (please complete the following information):**
 - OS and OS Version/extra details: All platforms
 - Eclipse Version from Help -> 2023-09 (it works in 2023-06)
 - CDT Version from Help -> About -> Installation Details -> Features tab 11.3.0 (works in 11.2.0)


**Additional context**

See [error log](https://github.com/eclipse-cdt/cdt/files/13169594/disassembly.log) for some exceptions that are observed.

With better exception handling (see https://github.com/eclipse-cdt/cdt/pull/604) the underlying exception for the items in the error log looks like:

```java
Caused by: org.eclipse.jface.text.BadLocationException: Offset > length: 79 > 0
	at org.eclipse.jface.text.ListLineTracker.getLineNumberOfOffset(ListLineTracker.java:141)
	at org.eclipse.jface.text.AbstractLineTracker.getLineNumberOfOffset(AbstractLineTracker.java:246)
	at org.eclipse.jface.text.AbstractDocument.getLineOfOffset(AbstractDocument.java:863)
	at org.eclipse.cdt.dsf.debug.internal.ui.disassembly.provisional.DisassemblyViewer.revealOffset(DisassemblyViewer.java:318)
	... 31 more
```

--- a/dsf/org.eclipse.cdt.dsf.ui/META-INF/MANIFEST.MF
+++ b/dsf/org.eclipse.cdt.dsf.ui/META-INF/MANIFEST.MF
@@ -3,7 +3,7 @@ Bundle-ManifestVersion: 2
 Bundle-Name: %pluginName
 Bundle-Vendor: %providerName
 Bundle-SymbolicName: org.eclipse.cdt.dsf.ui;singleton:=true
-Bundle-Version: 2.7.100.qualifier
+Bundle-Version: 2.7.200.qualifier
 Bundle-Activator: org.eclipse.cdt.dsf.internal.ui.DsfUIPlugin
 Bundle-Localization: plugin
 Require-Bundle: org.eclipse.ui;bundle-version="3.5.0",

--- a/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/internal/ui/disassembly/DisassemblyPart.java
+++ b/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/internal/ui/disassembly/DisassemblyPart.java
@@ -2126,8 +2126,10 @@ public abstract class DisassemblyPart extends WorkbenchPart
 		fPCHistory.clear();
 		fPendingPCUpdates.clear();
 		fFile2Storage.clear();
-		fDocument.clear();
+		var oldDocument = fDocument;
+		fDocument = createDocument();
 		fViewer.setDocument(fDocument, new AnnotationModel());
+		oldDocument.dispose();
 		if (fDebugSessionId != null) {
 			attachBreakpointsAnnotationModel();
 			attachExtendedPCAnnotationModel();

--- a/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/internal/ui/disassembly/model/DisassemblyDocument.java
+++ b/dsf/org.eclipse.cdt.dsf.ui/src/org/eclipse/cdt/dsf/debug/internal/ui/disassembly/model/DisassemblyDocument.java
@@ -33,14 +33,12 @@ import org.eclipse.cdt.debug.internal.ui.disassembly.dsf.IDisassemblyDocument;
 import org.eclipse.cdt.debug.internal.ui.disassembly.dsf.LabelPosition;
 import org.eclipse.cdt.dsf.debug.internal.ui.disassembly.SourcePosition;
 import org.eclipse.cdt.dsf.debug.internal.ui.disassembly.text.REDDocument;
-import org.eclipse.cdt.dsf.debug.internal.ui.disassembly.text.REDTextStore;
 import org.eclipse.core.resources.IStorage;
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.Path;
 import org.eclipse.debug.core.sourcelookup.containers.LocalFileStorage;
 import org.eclipse.jface.text.BadLocationException;
 import org.eclipse.jface.text.BadPositionCategoryException;
-import org.eclipse.jface.text.DefaultLineTracker;
 import org.eclipse.jface.text.DocumentRewriteSession;
 import org.eclipse.jface.text.DocumentRewriteSessionType;
 import org.eclipse.jface.text.IDocument;
@@ -127,16 +125,6 @@ public class DisassemblyDocument extends REDDocument implements IDisassemblyDocu
 		fFileInfoMap.clear();
 		fInvalidAddressRanges.clear();
 		fInvalidSource.clear();
-	}
-
-	/**
-	 * Clears all content and state.
-	 */
-	public void clear() {
-		dispose();
-		setTextStore(new REDTextStore());
-		setLineTracker(new DefaultLineTracker());
-		completeInitialization();
 	}
 
 	public AddressRangePosition[] getInvalidAddressRanges() {

